docs/rules/lines-between-class-members.md

```diff
@@ -0,0 +1,103 @@
+# require or disallow an empty line between class members (lines-between-class-members)
+
+This rule improves readability by enforcing lines between class members. It will not check empty lines before the first member and after the last member, since that is already taken care of by padded-blocks.
+
+## Rule Details
+
+Examples of **incorrect** code for this rule:
+
+```js
+/* eslint lines-between-class-members: ["error", "always"]*/
+class MyClass {
+  foo() {
+    //...
+  }
+  bar() {
+    //...
+  }
+}
+```
+
+Examples of **correct** code for this rule:
+
+```js
+/* eslint lines-between-class-members: ["error", "always"]*/
+class MyClass {
+  foo() {
+    //...
+  }
+
+  bar() {
+    //...
+  }
+}
+```
+
+### Options
+
+This rule has a string option and an object option.
+
+String option:
+
+* `"always"`(default) require an empty line after after class members
+* `"never"` disallows an empty line after after class members
+
+Object option:
+
+* `"exceptAfterSingleLine": "false"`(default) **do not** skip checking empty lines after singleline class members
+* `"exceptAfterSingleLine": "true"` skip checking empty lines after singleline class members
+
+Examples of **incorrect** code for this rule with the string option:
+
+```js
+/* eslint lines-between-class-members: ["error", "always"]*/
+class Foo{
+  bar(){}
+  baz(){}
+}
+
+/* eslint lines-between-class-members: ["error", "never"]*/
+class Foo{
+  bar(){}
+
+  baz(){}
+}
+```
+
+Examples of **correct** code for this rule with the string option:
+
+```js
+/* eslint lines-between-class-members: ["error", "always"]*/
+class Foo{
+  bar(){}
+
+  baz(){}
+}
+
+/* eslint lines-between-class-members: ["error", "never"]*/
+class Foo{
+  bar(){}
+  baz(){}
+}
+```
+
+Examples of **correct** code for this rule with the object option:
+
+```js
+/* eslint lines-between-class-members: ["error", "always", { exceptAfterSingleLine: true }]*/
+class Foo{
```

Comment:
Can we make this example more clear? It's not obvious how it's different from the default option.

Maybe something like:
```
class Foo{
  bar(){} // single line class member
  baz(){
    // multi line class member
  }

  qux(){}
}
```